File: C++/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(std::vector<int> &nums) {
        int sum = 0;
        int maxsum = nums[0];
        
        for(auto i : nums) {
            sum += i;
            if(sum>maxsum) maxsum = sum;
            if(sum<0) sum = 0;
        }
        return maxsum;
    }
};

File: Python/maximum-subarray.py
class Solution:
    # @param A, a list of integers
    # @return an integer
    def maxSubArray(self, A):
        S = 0
        M = A[0]
        
        for i in A:
            S += i
            if S>M: M = S
            if S<0: S = 0
        return M